crates/containerd-shim-wasmtime/src/instance.rs

```diff
@@ -36,13 +36,24 @@ impl Engine for WasmtimeEngine {
 
         log::info!("wasi context ready");
         let WasiEntrypoint { path, func } = ctx.wasi_entrypoint();
-        let path = path
-            .resolve_in_path_or_cwd()
-            .next()
-            .context("module not found")?;
 
-        log::info!("loading module from file {path:?}");
-        let module = Module::from_file(&self.engine, &path)?;
+        let module = match ctx.wasm_layers() {
+            [] => {
+                log::info!("loading module from file");
+                let path = path
+                    .resolve_in_path_or_cwd()
+                    .next()
+                    .context("module not found")?;
+
+                Module::from_file(&self.engine, path)?
+            }
+            [module] => {
+                log::info!("loading module wasm OCI layers");
+                Module::from_binary(&self.engine, &module.layer)?
+            }
+            [..] => bail!("only a single module is supported when using images with OCI layers"),
```

Comment:
Do we intend to support multiple images in an OCI artifact in the future?

Reply:
I was assuming that was there for components.

Reply:
The shims don't currently have the component features turned on, so wouldn't know how to handle multi components.  As a follow up we can enable multiple components